insights/v1alpha1/types_insights.go

```diff
@@ -0,0 +1,125 @@
+}
+
+type DataGatherSpec struct {
+	// dataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data. Valid values are "None" and "ObfuscateNetworking".
+	// When set to None the data is not obfuscated.
+	// When set to ObfuscateNetworking the IP addresses and the cluster domain name are obfuscated.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is None.
+	// +optional
+	DataPolicy DataPolicy `json:"dataPolicy"`
+	// disabledGatherers is a list of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// Run the following command to get the names of last active gatherers:
+	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
+	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +optional
+	DisabledGatherers []string `json:"disabledGatherers"`
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "None"
+	// IP addresses and cluster domain name are obfuscated
+	ObfuscateNetworking DataPolicy = "ObfuscateNetworking"
+	// Data gathering is running
+	Running State = "Running"
+	// Data gathering is completed
+	Completed State = "Completed"
+	// Data gathering failed
+	Failed State = "Failed"
+	// Data gathering is pending
+	Pending State = "Pending"
+)
+
+// dataPolicy declares valid data policy types
+// +kubebuilder:validation:Enum="";None;ObfuscateNetworking
+type DataPolicy string
+
+// state declares valid gathering state types
+// +kubebuilder:validation:Enum=Running;Completed;Failed
+type State string
+
+type DataGatherStatus struct {
+	// state reflects the current state of the data gathering process.
+	State State `json:"state,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	// +listType=atomic
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+	// startTime is the time when Insights data gathering started.
+	StartTime metav1.Time `json:"startTime,omitempty"`
+	// finishTime is the time when Insights data gathering finished.
+	FinishTime metav1.Time `json:"finishTime,omitempty"`
+	// relatedObjects is a list of resources which are useful when debugging or inspecting the data
+	// gathering Pod
+	RelatedObjects []configv1.ObjectReference `json:"relatedObjects,omitempty"`
+	// healthChecks provides basic information about active Insights health checks
+	// in a cluster.
+	// +listType=atomic
+	// +optional
+	HealthChecks []operatorv1.HealthCheck `json:"healthChecks,omitempty"`
```

Comment:
We don't normally allow cross API group references like this, @deads2k would you expect local copies of these types to be created?

Reply:
Yes this is a good point, I wanted to ask, but forgot.

Reply:
> We don't normally allow cross API group references like this, @deads2k would you expect local copies of these types to be created?

correct, we do not allow configv1 to depend on operatorv1.  If there's an off-the-shelf import analysis verification tool we can use to prevent it, I'm in favor.